modules/tensor_mechanics/src/materials/ComputeCrackedStress.C

```diff
@@ -73,7 +94,10 @@ void
 ComputeCrackedStress::initQpStatefulProperties()
 {
   _stress[_qp].zero();
-  _hist[_qp] = 0.0;
+  if (_precracked & (_c[_qp] > _c_threshold))
+    _hist[_qp] = _hist_init_factor * _gc_prop[_qp] / 2.0 / _l[_qp];
```

Comment:
What's the deal with _hist_init_factor? Are you inventing a method or following some reference? There are analytical solutions of the history variable -- apparently it depends on the specific phase-field fracture you use. For example the history variable is different when you use AT2 and AT1 models. The default value of your _hist_init_factor is 100, in some cases it will inject too much energy into the system and your initial crack will be too wide, and in some other cases it is not enough to enforce the irreversibility.

Reply:
> More and more literatures are switching from the history approach to a variational inequality approach to enforce the irreversibility. I think the phase-field fracture capabilities in MOOSE are also following that trend in deprecating the history approach. Recently several Bounds (for irreversibility see VariableOldValueBoundAux) were also added to the framework to support the variational inequality.

We support expanding MOOSE capabilities to accommodate this new approach, but we feel that the proposed changes have value for those currently using the history approach.  We have been using this approach in conjunction with the proposed changes for some time with good results.  

> What's the deal with _hist_init_factor? Are you inventing a method or following some reference? There are analytical solutions of the history variable -- apparently it depends on the specific phase-field fracture you use. For example the history variable is different when you use AT2 and AT1 models. The default value of your _hist_init_factor is 100, in some cases it will inject too much energy into the system and your initial crack will be too wide, and in some other cases it is not enough to enforce the irreversibility.

This method is based on the unpublished work of one of our lab members.  It is true that if `_hist_init_factor` is too small, the energy will be insufficient to stabilize the pre-existing crack, and it will heal.  We conduct tests for each case to identify the minimum value necessary to stabilize the crack.  

We have found that values between 100 and 1000 are typically sufficient.  We have not observed widening of the pre-existing crack when using values above this threshold (beyond the initial time step, during which the shape of `c` changes slightly to adapt to the initial value of `hist`).

Reply:
There has already been some work along this line -- see [Michael Borden's dissertation](https://repositories.lib.utexas.edu/handle/2152/ETD-UT-2012-08-6113) appendix B1. It seems that your (or your colleague's) approach is a simplified version of that one. Then my question becomes is this simplification really necessary as it introduces another heuristic damage threshold (0.99 is your default value I believe). Why not just use equations B.1 and B.2 in his dissertation?

Another issue here is that this equation only works when your local dissipation potential is quadratic in c. If the local dissipation potential is linear in c, then this equation will give you negative damage.

If your colleague wants to publish this work, I think he or she will get these questions from reviewers anyways...

Reply:
"Another issue here is that this equation only works when your local dissipation potential is quadratic in c. If the local dissipation potential is linear in c, then this equation will give you negative damage."

This `ComputeCrackedStress` is only implemented for AT2 (local dissipation is quadratic in c), so this approach should be fine.